docs/resources-overview.md

```diff
@@ -0,0 +1,112 @@
+Nitric provides cloud native building blocks that make it simple to declare the resources you need as part of your application code.
+
+The Nitric deployment engine will run through your code at deploy time and interpret resources that are declared as part of your application and create them in the cloud you are pushing to.
+
+## Rules
+
+There are a few rules to keep in mind when declaring nitric resources as part of your application
+
+### Don't declare resources in runtime code
+
+Nitric needs to be aware of resources at deploy time so they can be deployed appropriately. Declaring resources at runtime mean the resource won't be declared when deploying your application and therefore it will not be provisioned to the cloud.
+
+The Nitric deployment engine does not evaluate runtime code at deploy time as this could result in unintended behaviour/side-effects.
```

Comment:
```suggestion
The Nitric deployment engine does not evaluate runtime code at deploy time as this could result in unintended behaviours or side-effects.
```